app/src/main/java/com/nio/tcpdataflowexample/server/TcpServer.java

```diff
@@ -0,0 +1,29 @@
+package com.nio.tcpdataflowexample.server;
+
+import java.io.IOException;
+
+public abstract class TcpServer implements Server {

```

Comment:
What is the sense to split hostname and port on two different methods? Potentially the server will be valid if you don't put server port? I guess no.